### PR TITLE
Add additional VectorN implementations for some math functions.

### PR DESCRIPTION
**Pull request by bernardkwok**
Update #543 
Add vector2/3/4 for trig functions, ln,exp,sqrt.
 Note: tests need to be added but are temporary ignored by putting them on the whitelist.

### CODE REVIEW

File: source/MaterialXTest/GenOsl.cpp [hunk 1]
```diff
@@ -92,7 +92,7 @@ TEST_CASE("GenShader: OSL Implementation Check", "[genosl]")
     generatorSkipNodeTypes.insert("light");
     mx::StringSet generatorSkipNodeDefs;
 
-    GenShaderUtil::checkImplementations(context, generatorSkipNodeTypes, generatorSkipNodeDefs, 65);
+    GenShaderUtil::checkImplementations(context, generatorSkipNodeTypes, generatorSkipNodeDefs, 65+27);
```
Comment (feldstj) [inline source/MaterialXTest/GenOsl.cpp]: Any reason you're writing this as 65+27 instead of 92?

File: source/MaterialXTest/GenOsl.h [hunk 2]
```diff
@@ -56,7 +56,7 @@ class OslShaderGeneratorTester : public GenShaderUtil::ShaderGeneratorTester
         whiteList =
         {
             "ambientocclusion", "arrayappend", "backfacing", "screen", "curveadjust", "displacementshader",
-            "volumeshader", "IM_constant_", "IM_dot_", "IM_geomattrvalue"
+            "volumeshader", "IM_constant_", "IM_dot_", "IM_geomattrvalue",
```
Comment (feldstj) [inline source/MaterialXTest/GenOsl.h]: minor: why the ","?

File: source/MaterialXTest/GenOsl.cpp (hunk 1 above)
Comment (bernardkwok) [reply source/MaterialXTest/GenOsl.cpp]: Mostly to remember there are 27 tests to add :)

File: source/MaterialXTest/GenOsl.h (hunk 2 above)
Comment (bernardkwok) [reply source/MaterialXTest/GenOsl.h]: I had modified this before. Will revert.

**Comment by bernardkwok** [comment]
Versiion with tests in PR into master. Will merge from there.

**Pull request closed without merging**